src/cli.rs

```diff
@@ -331,11 +331,37 @@ impl Opt {
                 process::exit(1);
             }
         };
-        let root = root.unwrap_or_else(|| home.join(".sheldon"));
-        let config_file = config_file.unwrap_or_else(|| root.join("plugins.toml"));
-        let lock_file = lock_file.unwrap_or_else(|| root.join("plugins.lock"));
-        let clone_dir = clone_dir.unwrap_or_else(|| root.join("repos"));
-        let download_dir = download_dir.unwrap_or_else(|| root.join("downloads"));
+
+        let xdg_config_user = std::env::var_os("XDG_CONFIG_HOME").map(PathBuf::from);
+        let xdg_data_user = std::env::var_os("XDG_DATA_HOME").map(PathBuf::from);
+
+        // Note: "XDG_RUNTIME_DIR is not checked as it can be set by the
+        // system rather than the user, and cannot be relied upon to
+        // indicate a preference for XDG directory layout.
+        let using_xdg = xdg_data_user.is_some()
+            || xdg_config_user.is_some()
+            || std::env::var_os("XDG_CACHE_HOME").is_some()
+            || std::env::var_os("XDG_DATA_DIRS").is_some()
+            || std::env::var_os("XDG_CONFIG_DIRS").is_some();
+
+        let (config_dir, data_dir) = if using_xdg {
+            (
+                xdg_config_user
+                    .unwrap_or_else(|| home.join(".config"))
+                    .join("sheldon"),
+                xdg_data_user
+                    .unwrap_or_else(|| home.join(".local/share"))
+                    .join("sheldon"),
+            )
+        } else {
+            (home.join(".sheldon"), home.join(".sheldon"))
```

Comment:
I think you need to take into account `root` here (in case someone specifies it in the CLI or via environment variable). Something like the following:

```suggestion
            let dir = root
                .clone()
                .unwrap_or_else(|| home.clone())
                .join(".sheldon");
            (dir.clone(), dir)
```

Reply:
You're right. Amended.